apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,514 @@
+        {
+            return string.Concat(Enumerable.Repeat(character, length));
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("1", "1")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("NAN", "Nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("0.00", "0")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("0.00000009", "0.00000009000000")]
+        [InlineData("0.00000009000000", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void EqualsTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.Equals(number2);
+            Assert.Equal(true, result);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("1", "1")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("NAN", "Nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("0.00", "0")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("0.00000009", "0.00000009000000")]
+        [InlineData("0.00000009000000", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void CompareToEqualValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(0, result);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetComparisionData))]
+        public void CompareToTest(string input1, string input2, int expectedResult)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(expectedResult, Math.Sign(result));
+        }
+
+        [Fact]
+        public void CompareTo_NegativeZero()
+        {
+            // These really are the same value... we don't differentiate.
+            var regularZero = PgNumeric.Zero;
+            var negativeZero = PgNumeric.Parse("-0");
+            Assert.Equal(0, regularZero.CompareTo(negativeZero));
+            Assert.Equal(0, (IComparable)regularZero.CompareTo(negativeZero));
+            Assert.Equal(0, negativeZero.CompareTo(regularZero));
+            Assert.Equal(0, (IComparable)negativeZero.CompareTo(regularZero));
+        }
+
+        [Fact]
+        public void CompareTo_Null()
+        {
+            IComparable minValue = PgNumeric.MinValue;
+            Assert.InRange(minValue.CompareTo(null), 1, int.MaxValue);
```

Comment:
Use Math.Sign is for consistency here.
```suggestion
            Assert.Equal(1, Math.Sign(minValue.CompareTo(null)));
```